smac/utils/validate.py

```diff
@@ -0,0 +1,371 @@
+            dictionary mapping instances to a list of tuples of already used
+            seeds and the configs that this inst-seed-pair has been evaluated
+            on, sorted by the number of configs
+        """
+        # We want to reuse seeds that have been used on most configurations
+        # To this end, we create a dictionary as {instances:{seed:[configs]}}
+        # Like this we can easily retrieve the most used instance-seed pairs to
+        # minimize the number of runs to be evaluated
+        inst_seed_config = {}
+        if runhistory:
+            relevant = dict()
+            for key in runhistory.data:
+                if (runhistory.ids_config[key.config_id] in configs
+                        and key.instance_id in insts):
+                    relevant[key] = runhistory.data[key]
+
+            # Change data-structure to {instances:[(seed1, (configs)), (seed2, (configs), ... ]}
+            # to make most used seed easily accessible, we sort after length of configs
+            for key in relevant:
+                inst, seed = key.instance_id, key.seed
+                config = runhistory.ids_config[key.config_id]
+                if inst in inst_seed_config:
+                    if seed in inst_seed_config[inst]:
+                        inst_seed_config[inst][seed].append(config)
+                    else:
+                        inst_seed_config[inst][seed] = [config]
+                else:
+                    inst_seed_config[inst] = {seed : [config]}
+
+            inst_seed_config = {i :
+                                sorted([(seed, tuple(inst_seed_config[i][seed]))
+                                        for seed in inst_seed_config[i]],
+                                       key=lambda x: len(x[1]))
+                                for i in inst_seed_config}
+        return inst_seed_config
+
+
+    def _get_configs(self, mode):
+        """
+        Return desired configs
+
+        Parameters
+        ----------
+        mode : string
+            from [def, inc, def+inc, time, all], time means eval at 2^0, 2^1, 2^2, ...
+
+        Returns
+        -------
+        configs: list<Configuration>
+            list with desired configurations
+        """
+        # Add desired configs
+        configs = []
+        mode = mode.lower()
+        if mode not in ['def', 'inc', 'def+inc', 'time', 'all']:
+            raise ValueError("%s not a valid option for config_mode in validation."
+                             %mode)
+        if mode == "def" or mode == "def+inc":
+            configs.append(self.scen.cs.get_default_configuration())
+        if mode == "inc" or mode == "def+inc":
+            configs.append(self.traj[-1]["incumbent"])
+        if mode == "time":
+            # add configs at evaluations 2^1, 2^2, 2^3, ...
+            counter = 1
+            for entry in self.traj[:-1]:
+                if (entry["evaluations"] >= counter and
```

Comment:
please use `wallclock_time` here instead of `evaluations`.
and please start with the first entry in the traj
and then continue with 2^-4, 2^-3, 2^-2, ...